platform/security/core/security-core-impl/src/main/java/ddf/security/samlp/impl/LogoutMessageImpl.java

```diff
@@ -191,7 +184,31 @@ public LogoutResponse buildLogoutResponse(
         SamlProtocol.createIssuer(issuerOrEntityId),
         SamlProtocol.createStatus(statusCodeValue),
         inResponseTo,
-        id);
+        "_" + UUID.randomUUID().toString());
+  }
+
+  @Override
+  public LogoutResponse buildLogoutResponse(
+      String issuerOrEntityId,
+      String topLevelStatusCode,
+      String secondLevelStatusCode,
+      String inResponseTo) {
+    if (issuerOrEntityId == null) {
+      throw new IllegalArgumentException("Issuer cannot be null");
+    }
+    if (topLevelStatusCode == null || secondLevelStatusCode == null) {
+      throw new IllegalArgumentException("Status Codes cannot be null");
+    }
+
+    Status status = SamlProtocol.createStatus(topLevelStatusCode);
+    StatusCode statusCode = SamlProtocol.createStatusCode(secondLevelStatusCode);
+    status.getStatusCode().setStatusCode(statusCode);
```

Comment:
❓ Not `status.setStatusCode(statusCode);`?

Reply:
No, we want to set the status code on the status code node:
```
<saml2p:Status>
    <saml2p:StatusCode Value="urn:oasis:names:tc:SAML:2.0:status:Success">
        <saml2p:StatusCode Value="urn:oasis:names:tc:SAML:2.0:status:PartialLogout"/>
    </saml2p:StatusCode>
</saml2p:Status>
```

Reply:
No, we want to set the `<StatusCode>` of the first `<StatusCode>`
```
<saml2p:Status>
    <saml2p:StatusCode Value="urn:oasis:names:tc:SAML:2.0:status:Success">
        <saml2p:StatusCode Value="urn:oasis:names:tc:SAML:2.0:status:PartialLogout"/>
    </saml2p:StatusCode>
</saml2p:Status>
```